### DISCUSSION TITLE
Unable to set destination route when connection to channel

### DISCUSSION THREAD

**Discussion by szbenceg** [rsocket/rsocket-js#248] [Q&A] [answered]
I am new to Rsocket and reactive Websocket and I develop an application where I need multiple channels, where my clients can subscribe. Until now I tried 2 solutions but unfortunately non of them workd as I expected.

1. I implemented a reactive Websocket server with Spring where I stored every client connection in a FluxSink list, but unfortunately with that solution I couldn't manage to implement channels, and than I read @OlegDokuka answer [here](https://stackoverflow.com/questions/53812515/webflux-websocketclient-how-to-send-multiple-requests-in-same-sessiondesign-cl) and I believed him not trying to reimplement a protocol I am not so hardcore 😢 .

2. So I tried RSocket because I read that with this protocol it is quite easy to create bidirectional channels, but unfortunately I cannot connect to them

I have a Controller with this @MessageMapping

```
@MessageMapping("my-channel")
public Flux<String> channel(final Flux<String> message) {
  log.info("Received stream-stream (channel) request... ");

  return message
        .doOnNext(m -> log.info("Request: ", m))
        .doOnCancel(() -> log.warn("Client cancelled the channel"))
        .switchMap(setting -> Flux.interval(Duration.ofSeconds(10)).map(index -> "Love spring if it works"))
        .log()
        ;
 }
```
Unfortunately when I try to connect from my react native client the Destination route is alwayes ' '. Of course if I change @MessageMapping("my-channel") -> @MessageMapping("") it works perfectly but this solution is not appropriate for me.

This is my client connection code:
```
const connector = new RSocketConnector({
  setup: {
    keepAlive: 100,
    lifetime: 10000,
    metadataMimeType: "application/json",
    dataMimeType: "message/x.rsocket.routing.v0"
  },
  transport: new WebsocketClientTransport({
    url: "ws://192.168.0.107:7000",
    wsCreator: (url) => new WebSocket(url) as any,
  }),
});

const rsocket = await connector.connect();

await new Promise((resolve, reject) => {
  const requester = rsocket.requestChannel(
    {
      data: undefined,
      metadata: Buffer.concat([
        Buffer.from(String.fromCharCode('my-channel'.length)),
        Buffer.from('my-channel'),
      ]),
    },
    1,
    false,
    {
      onError: (e) => reject(e),
      onNext: (payload, isComplete) => {
        console.log(
          `payload[data: ${payload.data}; metadata: ${payload.metadata}]|${isComplete}`
        );

        requester.request(1);

        if (isComplete) {
          resolve(payload);
        }
      },
      onComplete: () => {
        resolve(null);
      },
      onExtension: () => { },
      request: (n) => {
        console.log(`request(${n})`);
        requester.onNext(
          {
            data: Buffer.from("Message"),
          },
          true
        );
      },
      cancel: () => { },
    }
  );
}).catch(e => {
  console.log("-------------ERRORRRRRR-------------")
  console.error(e);
});
```
The error message is this:

```
-------------ERRORRRRRR-------------
ERROR  [Error: Destination '' does not support REQUEST_CHANNEL. Supported interaction(s): [METADATA_PUSH, SETUP]]
```

What am I doing wrong?
Thank you for your help in advance. 😄

**Comment by OlegDokuka** [comment, marked as answer]
You have specified data and metadata mime types on the client nt side incorrectly.

Actual

```
metadataMimeType: "application/json",
dataMimeType: "message/x.rsocket.routing.v0"
```

Have to be

```
dataMimeType: "application/json",
    metadataMimeType: "message/x.rsocket.routing.v0"
```

**Reply by szbenceg** [reply]
Thank you so much. I can't believe I spent with this issue over 20 hours, and that was all the problem.
